hyperspy/tests/signal/test_eds_tem.py

```diff
@@ -157,15 +159,49 @@ def setUp(self):
 
         s.set_elements(elements)
         s.add_lines(xray_lines)
+        s.axes_manager[0].scale = 0.5
+        s.axes_manager[1].scale = 0.5
         self.signal = s
 
     def test_quant_lorimer(self):
         s = self.signal
+        method = 'CL'
         kfactors = [1, 2.0009344042484134]
+        composition_units = 'weight'
         intensities = s.get_lines_intensity()
-        res = s.quantification(intensities, kfactors)
-        nt.assert_true(np.allclose(res[0].data, np.array(
-                       [22.70779, 22.70779]), atol=1e-3))
+        res = s.quantification(intensities, method, kfactors, composition_units)
+        np.testing.assert_allclose(res[0].data, np.array([[22.70779, 22.70779],
+                    [22.70779, 22.70779]]), atol=1e-3)
+
+    def test_quant_zeta(self):
+        s = self.signal
+        method = 'zeta'
+        compositions_units = 'weight'
+        factors = [20, 50]
+        intensities = s.get_lines_intensity()
+        res = s.quantification(intensities, method, factors, compositions_units)
+        np.testing.assert_allclose(res[1].data, np.array(
+                [[ 2.7125736e-03,   2.7125736e-03],
+                [  2.7125736e-03,   2.7125736e-03]]), atol=1e-3)
+        np.testing.assert_allclose(res[0][1].data, np.array(
+                [[ 80.962287987,   80.962287987],
+                [  80.962287987,   80.962287987]]), atol=1e-3)
```

Comment:
I'm pretty confident that I've gone through it several times now. I'll double check with real data again next week. I agree it's better with realistic values but I was trying to use the simple spectrum Pierre set up for the k-factor quant test. Probe current, cross sections and dwell time are all realistic. Could probably increase the dwell time to make the atom counts look more realistic if that's preferable?

@siriagus a second test on the z-factors would be very helpful as I have to experimental data to check that.

> On 8 Apr 2016, at 14:42, Duncan Johnstone <notifications@github.com> wrote:
> 
> In hyperspy/tests/signal/test_eds_tem.py:
> 
> > +        np.testing.assert_allclose(res[0].data, np.array([[22.70779, 22.70779],
> > +                    [22.70779, 22.70779]]), atol=1e-3)
> > +
> > +    def test_quant_zeta(self):
> > +        s = self.signal
> > +        method = 'zeta'
> > +        compositions_units = 'weight'
> > +        factors = [20, 50]
> > +        intensities = s.get_lines_intensity()
> > +        res = s.quantification(intensities, method, factors, compositions_units)
> > +        np.testing.assert_allclose(res[1].data, np.array(
> > +                [[ 2.7125736e-03,   2.7125736e-03],
> > +                [  2.7125736e-03,   2.7125736e-03]]), atol=1e-3)
> > +        np.testing.assert_allclose(res[0][1].data, np.array(
> > +                [[ 80.962287987,   80.962287987],
> > +                [  80.962287987,   80.962287987]]), atol=1e-3)
> Ok - @siriagus could you do us a favour and check this on some of the data that you know well from your thesis? Just to double check everything is working correctly? I don't have any way to do this myself without a lot of effort really.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>